services/service-runner/app/service/service-runner.ts

```diff
@@ -92,9 +92,27 @@ export class LogStream extends stream.Writable {
  */
 async function _getStacCatalogs(dir: string): Promise<string[]> {
   const s3 = objectStoreForProtocol('s3');
-  return (await s3.listObjectKeys(dir))
-    .filter((fileKey) => fileKey.match(/catalog\d*.json/))
-    .map((fileKey) => `s3://${env.artifactBucket}/${fileKey}`);
+  // check to see if there is a batch-catalogs.json file and read it if so
+  const batchCatalogsJsonUrl = `${dir}batch-catalogs.json`;
+  if (await s3.objectExists(batchCatalogsJsonUrl)) {
+    const batchCatalogs = await s3.getObjectJson(batchCatalogsJsonUrl) as string[];
+    const result =  batchCatalogs.map(filename => `${dir}${filename}`);
+    return result;
+  }
+
+  // otherwise retrieve the keys from the bucket that are of the form catalog*.json,
+  // and sort them by index number
+  const urls = (await s3.listObjectKeys(dir))
+    .filter((fileKey) => fileKey.match(/catalog\d*.json$/))
+    .map((fileKey) => `${dir}${fileKey}`);
+  const fileNumRegex = /.*catalog(\d+)\.json$/;
```

Comment:
If there's only one catalog the file is named catalog.json (no digits in it). If this doesn't match anything will we still return the right thing?

I'm not sure if this code will get reached because I thought we might have a batchCatalogs.json file in all cases so maybe it doesn't matter, but we should test this logic.

Reply:
as you noted below I have a test for this